test/core/utils/preload.js

```diff
@@ -1,81 +1,54 @@
 describe('axe.utils.preload', function() {
 	'use strict';
 
-	it('should return a queue', function() {
-		var actual = axe.utils.preload({});
-		assert.isObject(actual);
-		assert.containsAllKeys(actual, ['then', 'defer', 'catch']);
-	});
+	var isPhantom = window.PHANTOMJS ? true : false;
+	var fixture = document.getElementById('fixture');
 
-	it('should ensure queue is defer(able)', function(done) {
-		var options = {
-			preload: false
-		};
-		var actual = axe.utils.preload(options);
-		actual
-			.defer(function(res, rej) {
-				assert.isFunction(rej);
-				res(true);
-				done();
-			})
-			.catch(function(error) {
-				done(error);
-			});
+	before(function() {
+		if (isPhantom) {
+			this.skip(); // if `phantomjs` -> skip `suite`
+		}
+		axe._tree = axe.utils.getFlattenedTree(fixture);
 	});
 
-	it('should ensure queue is then(able)', function(done) {
-		var options = {
-			preload: false
-		};
-		var actual = axe.utils.preload(options);
-		actual
-			.then(function(results) {
-				assert.isDefined(results);
-				done();
-			})
-			.catch(function(error) {
-				done(error);
-			});
+	it('returns a Promise', function() {
+		var actual = axe.utils.preload({});
+		assert.isTrue(
+			Object.prototype.toString.call(actual) === '[object Promise]'
```

Comment:
This won't work if the Promise is polyfilled in the test environment (such as IE). Would be better to do `actual instanceof Promise`

Reply:
The [algorithm used here](https://github.com/then/is-promise/blob/master/index.js) is downloaded 6.5 million times a week. I suggest we use it rather than `instanceof` or `toString()` checks.

Reply:
This is a simple assertion, so I would rather leave it as is.